src/server/db_slice.h

```diff
@@ -468,8 +469,21 @@ class DbSlice {
     }
   };
 
-  // the table that maps keys to the clients that are tracking them.
-  absl::flat_hash_map<std::string, absl::flat_hash_set<facade::Connection::WeakRef, Hash>>
+  using HashSetAllocator = PMR_NS::polymorphic_allocator<facade::Connection::WeakRef>;
+
+  using AllocatorType = PMR_NS::polymorphic_allocator<std::pair<
+      std::string,
+      absl::flat_hash_set<facade::Connection::WeakRef, Hash,
```

Comment:
maybe define the hash set above as well

Reply:
added. 

The current way for manipulating these types to make them correct enough for supporting user provided allocator is not very elegant... in the future we should try to provide some sort of wrappers to hide these details. The code now is very hard to read what exactly we are doing here...